File: Level1/예산.cpp
#include <iostream>
#include <stdio.h>
#include <string>
#include <vector>
#include <algorithm>
using namespace std;

int solution(vector<int> d, int budget) {
    int answer = 0;
    int tmp = 0;
    
    sort(d.begin(),d.end());
    for(int i = 0; i < d.size(); i++)
    {
        if(tmp + d[i] <= budget)
        {
            tmp += d[i];
            answer++;
        }
    }
    return answer;
}

File: Level1/예산.js
const binarySearch = (d, mid, budget) => {
    let sum = 0;
    
    for(let i = 0; i <= mid; i++){
        sum += d[i];
        
        if(sum > budget){
            return true;
        }
    }
    
    return false;
}

const solution = (d, budget) => {
    let front = 0;
    let back = d.length - 1;
    
    d.sort((a, b) => a - b);
    
    while(front <= back){
        const mid = Math.floor((front + back) / 2);
        
        if(binarySearch(d, mid, budget)){
            back = mid - 1;
        } else{
            front = mid + 1;
        }
    }
    
    return front;
}